### PR TITLE
Add script for determining changed apps

### PR DESCRIPTION
**Pull request by theodur**
## Description
In order to build only changed apps in GHA, we need a way for the `continuous-integration` workflow to know which apps to build and which unit tests to run. The work for determining which apps to build and which unit tests to run was initially split into two tickets, but because the approaches were similar, we decided to combine them into a single script. This PR contains a combined script based off the work that @alexandec and I did in our separate tickets.

`get-changed-apps.js` will generate a list of changed apps from the changed files in the environment, and output that list as a comma-delimited string of either app entry names or relative paths, depending on the option used. All changed files must be part of an app on the [allow-list](https://github.com/department-of-veterans-affairs/vets-website/blob/master/config/single-app-build.json), otherwise an empty string will be returned.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#31119
department-of-veterans-affairs/va.gov-team#31142

## Testing done
Tested in previous commits along with written unit tests. Most recent commit of only changed apps build: https://github.com/department-of-veterans-affairs/vets-website/actions/runs/1388525257

## Acceptance criteria
- [x] The script can determine changed apps from changed files in a branch.
- [x] The script can be used to determine which apps to build and which unit tests to run in GHA.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/github-actions/get-changed-apps.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      return '';
+    }
+  }
+
+  return [...new Set(allowedApps)].join(',');
+};
+
+if (process.env.CHANGED_FILE_PATHS) {
+  const changedFiles = process.env.CHANGED_FILE_PATHS.split(' ').filter(
+    filePath => filePath.startsWith('src/applications'),
```
Comment (alexandec) [inline script/github-actions/get-changed-apps.js:81]: Isn't this filter left over from our testing?

File: script/github-actions/get-changed-apps.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (alexandec) [reply script/github-actions/get-changed-apps.js:1]: We could potentially disable console linting only for the line where console.log is used

File: script/github-actions/get-changed-apps.js (hunk 1 above)
Comment (theodur) [reply script/github-actions/get-changed-apps.js]: Ohh shoot, yeah that needs to be removed. Nice catch, just pushed a commit for that.

Comment (alexandec) [reply script/github-actions/get-changed-apps.js]: A higher-level integration test would have caught it, but that would be more involved to create. Still not totally convinced it's worthwhile

**Comment by alexandec** [comment]
Looks good!

**Review by alexandec — APPROVED** [review]

File: script/github-actions/get-changed-apps.js (hunk 1 above)
Comment (theodur) [reply script/github-actions/get-changed-apps.js]: That's true. If or when the implementation gets more complex, that could be useful.

**Pull request merged**